### PR TITLE
core: Deno refactor

### PR DESCRIPTION
**Pull request by spolu**
## Description

Introduces JSExecutor and JSClient.

The deno JSRuntime expects to run in his own dedicated thread. This is something that we can't really ensure with the tokio runtime (either we're on a worker (shared) or on a spawn_blocking worker (shared as well)).

The JSExecutor is initialized on the main program, spawns a std:thread (a real thread) and initialized the v8 platform on it. It is then in charge of spawning a new thread for each request it receives to create a new JSRuntime (isolation) and use it to execute the JS code.

The interaction between JSExecutor and JSClient operates on an mpsc channel (infinite buffer so non blocking send) to which we send the request and a oneshot tokio channel to send back the response. The resulting interface is very simple since the JSClient is stored in a singleton and can be cloned from anywhere safely. 

Confirmed the identified memory leak has been solved by this. Hopefully this was the real leak. In any case we're back on recent deno \o/

## Risk

High given how core it is to the execution of Dust. But since this is much cleaner and well tested

## Deploy Plan

- deploy `core`

### CODE REVIEW

File: core/src/deno/js_executor.rs [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+        let result = {
+            let scope = &mut runtime.handle_scope();
+            let local = v8::Local::new(scope, value);
+            // Deserialize a `v8` object into a Rust type using `serde_v8`,
+            // in this case deserialize to a JSON `Value`.
+            serde_v8::from_v8::<ValueWithLogs>(scope, local)
+        };
+
+        drop(runtime);
```
Comment (fontanierh) [inline core/src/deno/js_executor.rs:233]: We should remove IMO otherwise we'll have drops everywhere soon

File: core/src/deno/js_executor.rs [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+            let guard = JS_CLIENT.read();
+            guard.clone()
+        };
+        match js_client {
+            Some(c) => Ok(c),
+            None => Err(anyhow!("JSExecutor not initialized")),
+        }
+    }
+
+    // JSExecutor must be created ans started on the main thread of the Rust program. The returned
```
Comment (flvndvd) [inline core/src/deno/js_executor.rs:155]: `nit`
```suggestion
    // JSExecutor must be created and started on the main thread of the Rust program. The returned
```

File: core/src/providers/google_ai_studio.rs [hunk 3]
```diff
@@ -624,6 +624,9 @@ impl LLM for GoogleAiStudioLLM {
                 //  or set it as the content.
                 if let Some(t) = p.text.as_ref() {
                     content = content.map(|c| c + t).or_else(|| Some(t.clone()));
+                    // content = content
```
Comment (fontanierh) [inline core/src/providers/google_ai_studio.rs:627]: nit

File: core/src/deno/js_executor.rs [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+        tokio::task::spawn(async move {
+            tokio::time::sleep(timeout).await;
+            handle.terminate_execution();
+        });
+
+        #[allow(deprecated)]
+        let value = runtime.resolve_value(promise).await?;
+
+        // let ok = runtime.v8_isolate().cancel_terminate_execution();
```
Comment (fontanierh) [inline core/src/deno/js_executor.rs:222]: nit

**Review by fontanierh — APPROVED** [review]
nice

File: core/src/providers/google_ai_studio.rs (hunk 3 above)
Comment (flvndvd) [inline core/src/providers/google_ai_studio.rs]: ✂️

**Review by flvndvd — APPROVED** [review]
\o/ \o/ \o/

**Pull request merged**